### ISSUE TITLE
The BN running mean&var with torch.utils.checkpoint.checkpoint

### ISSUE THREAD

**Issue by ljn114514** [gpleiss/efficient_densenet_pytorch#73]
How do you deal with the bn running mean/variance? Because the BatchNorm would be calculated twice (once during the forward pass and once during recomputation in the backward pass), and the running mean&var would updated twice.

**Comment by gpleiss** [comment]
This is a good point. Ideally, PyTorch's batch norm layers should be smart enough to update the running mean/var appropriately with the checkpointing operation.

If this is not the case, then you should raise an issue with PyTorch, since the checkpointing/batch norm layers are part of their library, not this library.

**Comment by ljn114514** [comment]
Ok, thanks for your reply
